### PR TITLE
Add templates

### PR DESCRIPTION
**Pull request by ubergesundheit**
Barebones `flux-app` to install flux onto a workload cluster along with a GitSource and Kustomization CR

At this time, this does not support flux way of encrypting secrets in the repository flux watches

### CODE REVIEW

**Review by glitchcrab — APPROVED** [review]
initial look-over seems sensible - crd install manifests look good and the value schema doesn't have any null types :+1:

File: helm/flux-app/templates/crd-install/crd-np.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  # allow egress traffic to the Kubernetes API
+  egress:
+  - ports:
+    - port: 443
+      protocol: TCP
+    # legacy port kept for compatibility
+    - port: 6443
+      protocol: TCP
+    to:
+    {{- range tuple "10.0.0.0/8" "172.16.0.0/12" "192.168.0.0/16" "100.64.0.0/10" }}
```
Comment (piontec) [inline helm/flux-app/templates/crd-install/crd-np.yaml:29]: Is this coming from upstream? As we don't need NetPol to talk to k8s API, AFAIR it allows the traffic. And unless you create at least 1 NetPol in your netspace, the default behavior is "allow all"

File: helm/flux-app/templates/crd-install/crd-job.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: {{ include "crdInstall" . | quote }}
+        {{- include "labels.selector" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "crdInstall" . }}
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 2000
```
Comment (piontec) [inline helm/flux-app/templates/crd-install/crd-job.yaml:25]: Just checking: is this correct for a default (I presume) GID?

**Review by piontec — APPROVED** [review]
Haven't read CRDs and such, got just a few questions about some values.

File: helm/flux-app/templates/crd-install/crd-np.yaml (hunk 1 above)
Comment (ubergesundheit) [reply helm/flux-app/templates/crd-install/crd-np.yaml:29]: Copied this blindly from prometheus-operator-app.. 

@glitchcrab wdyt?

File: helm/flux-app/templates/crd-install/crd-job.yaml (hunk 2 above)
Comment (ubergesundheit) [reply helm/flux-app/templates/crd-install/crd-job.yaml:25]: Just checked: The uid and gid does not exist in the container image, but it seems this works nevertheless.

The values are also copied from prometheus-operator-app, but I think the values were not specific to the prometheus-operator-app chart.

**Pull request merged**